### DISCUSSION TITLE
Trying to write onto the bus | Getting EOF Errors

### DISCUSSION THREAD

**Discussion by CatOfSchroedinger** [canboat/canboat#231] [Q&A] [answered]
Hey, I am currently trying to send NMEA2000 Messages via Canboat onto the NMEA Network of a boat as a part of my Bachelor thesis. I`m using a Raspberry Pi 4 Model B 4GB with the newest versions of Openplotter on a 32GB SD-Card and Canboat on it. The Raspberry Pi is connected with a NGT-1 USB to the Network, with another NGT-1-USB connected to my laptop trying to spectate the messages I sent onto the NMEA2000. Reading the bus works fine using Canboat on the Raspberry, but if I try using the following code:

echo 2012-06-17-15:02:11.000,6,59904,0,255,3,14,f0,01 | actisense-serial /dev/ttyUSB0

I get a few lines of read NMEA2000 messages followed by the Error Message "FATAL 2021-03-31T08:20:25.366Z [actisense-serial] EOF on reading stdin". 

It would be very nice, if anyone could tell me what I am missing and have to change so I can send properly onto the bus.

Best regards,
Jannis

**Comment by keesverruijt** [comment, marked as answer]
This is expected behavior. If this does not send data to the bus, use 

https://github.com/canboat/canboat/wiki/n2kd_monitor

which shows how to do this (read the entire page!)

**Reply by keesverruijt** [reply]
@CatOfSchroedinger Sorry for the late reply, I was out sailing...
